container/webui/docker-compose.yaml

```diff
@@ -1,88 +1,70 @@
 version: '3.8'
-services:
-  data:
-    image: openqa_data
-    build: ../openqa_data/
-    restart: always
-    volumes:
-      - ./workdir/data/factory:/data/factory
-      - ./workdir/data/tests:/data/tests
 
+services:
   scheduler:
     image: openqa_webui
-    volumes:
-      - ./workdir/data/factory:/data/factory
-      - ./workdir/data/tests:/data/tests
-      - ./conf:/data/conf
+    volumes: &volumes
+      - ./workdir/data/factory/iso:/data/factory/iso:rw
+      - ./workdir/data/factory/hdd:/data/factory/hdd:rw
+      - ./workdir/data/factory/other:/data/factory/other:rw
+      - ./workdir/data/factory/tmp:/data/factory/tmp:rw
+      - ./workdir/data/testresults:/data/testresults:rw
+      - ./workdir/data/tests:/data/tests:ro
+      - ./conf:/data/conf:ro
     environment:
       MODE: "scheduler"
       MOJO_LISTEN: "http://0.0.0.0:9529"
     depends_on:
-      - data
       - db
 
   websockets:
     image: openqa_webui
     build: .
     ports:
       - "9527:9527"  # worker connection
-    volumes:
-      - ./workdir/data/factory:/data/factory
-      - ./workdir/data/tests:/data/tests
-      - ./conf:/data/conf
+    volumes: *volumes
     environment:
       MODE: "websockets"
       MOJO_LISTEN: "http://0.0.0.0:9527"
     depends_on:
-      - data
       - db
 
   gru:
     image: openqa_webui
-    build: .
-    volumes:
-      - ./workdir/data/factory:/data/factory
-      - ./workdir/data/tests:/data/tests
-      - ./conf:/data/conf
+    build: .    
+    volumes: *volumes
     environment:
       MODE: "gru"
     depends_on:
-      - data
       - db
 
   livehandler:
     image: openqa_webui
     build: .
     ports:
       - "9528:9528"  # handle live view
-    volumes:
-      - ./workdir/data/factory:/data/factory
-      - ./workdir/data/tests:/data/tests
-      - ./conf:/data/conf
+    volumes: *volumes
     environment:
       MODE: "livehandler"
       MOJO_LISTEN: "http://0.0.0.0:9528"
     depends_on:
-      - data
       - db
 
   webui:
     image: openqa_webui
     build: .
     ports:
       - "9526"
-    volumes:
-      - ./workdir/data/factory:/data/factory
-      - ./workdir/data/tests:/data/tests
-      - ./conf:/data/conf
+    volumes: *volumes
     environment:
       MODE: "webui"
       MOJO_LISTEN: "http://0.0.0.0:9526"
     depends_on:
-      - data
       - db
     deploy:
       replicas: ${OPENQA_WEBUI_REPLICAS}
+    entrypoint: "sh -c 'chmod -R a+rw /data/{factory,testresults}; /root/run_openqa.sh'"
```

Comment:
Those are directories so it would make sense to add `rwx` permissions instead of just `rw`.

Reply:
Better use `a+rwX`

Reply:
done